src/source.jl

```diff
@@ -44,23 +44,54 @@ end
 """
     ZlibInflateInputStream(input[; <keyword arguments>])
 
-Construct a zlib inflate input stream to decompress gzip/zlib data.
+Construct a zlib inflate input stream to decompress gzip data.
 
 # Arguments
 * `input`: a byte vector, IO object, or BufferedInputStream containing compressed data to inflate.
 * `bufsize::Integer=8192`: input and output buffer size.
-* `raw::Bool=falso`: if true, data is raw compress data, without zlib metadata
-* `gzip::Bool=true`: if true, data is gzip compressed; if false, zlib compressed.
 * `reset_on_end::Bool=true`: on stream end, try to find the start of another stream.
 """
-function ZlibInflateInputStream(input; bufsize::Integer=8192, raw::Bool=false,
-                                gzip::Bool=true, reset_on_end::Bool=true)
+function ZlibInflateInputStream(input; bufsize::Integer=8192, 
+                                reset_on_end::Bool=true)
     return BufferedInputStream(
-        InflateSource(input, bufsize, raw, gzip, reset_on_end),
+        InflateSource(input, bufsize, false, false, reset_on_end),
```

Comment:
I think this will break some packages that depende on the auto-format-detection mechanism of zlib. For example, see [Requests.jl](https://github.com/JuliaWeb/Requests.jl/blob/cf778031780fe006844103b15dff6e9e85166305/src/Requests.jl#L316), which uses `ZlibInflateInputStream` and will decompress both gzip and deflate. I'm not sure how many packages will be affected by this change.